includes/database/queries/class-order.php

```diff
@@ -164,19 +164,161 @@ public function __construct( $query = array() ) {
+	 *     @type string       $currency             Limit results to those affiliated with a given currency. Default empty.
+	 *     @type array        $currency__in         Array of currencies to include affiliated orders for. Default empty.
+	 *     @type array        $currency__not_in     Array of currencies to exclude affiliated orders for. Default empty.
+	 *     @type string       $payment_key          Limit results to those affiliated with a given payment key. Default empty.
+	 *     @type array        $payment_key__in      Array of payment keys to include affiliated orders for. Default empty.
+	 *     @type array        $payment_key__not_in  Array of payment keys to exclude affiliated orders for. Default empty.
+	 *     @type array        $date_query           Query all datetime columns together. See WP_Date_Query.
+	 *     @type array        $date_created_query   Date query clauses to limit orders by. See WP_Date_Query.
+	 *                                              Default null.
+	 *     @type array        $date_completed_query Date query clauses to limit orders by. See WP_Date_Query.
+	 *                                              Default null.
+	 *     @type bool         $count                Whether to return a order count (true) or array of order objects.
+	 *                                              Default false.
+	 *     @type string       $fields               Item fields to return. Accepts any column known names
+	 *                                              or empty (returns an array of complete order objects). Default empty.
+	 *     @type int          $number               Limit number of orders to retrieve. Default 100.
+	 *     @type int          $offset               Number of orders to offset the query. Used to build LIMIT clause.
+	 *                                              Default 0.
+	 *     @type bool         $no_found_rows        Whether to disable the `SQL_CALC_FOUND_ROWS` query. Default true.
+	 *     @type string|array $orderby              Accepts 'id', 'number', 'status', 'user_id', 'customer_id', 'email', 'gateway'
+	 *                                              'payment_key', 'date_created', 'date_completed', 'user_id__in', 'customer_id__in'.
+	 *                                              'email__in', 'gateway__in', 'payment_key__in'.
+	 *                                              Also accepts false, an empty array, or 'none' to disable `ORDER BY` clause.
+	 *                                              Default 'id'.
+	 *     @type string       $order                How to order retrieved orders. Accepts 'ASC', 'DESC'. Default 'DESC'.
+	 *     @type string       $search               Search term(s) to retrieve matching orders for. Default empty.
+	 *     @type bool         $update_cache         Whether to prime the cache for found orders. Default false.
+	 * }
+	 */
+	public function query( $query ) {
```

Comment:
This parameter needs to be changed to: `$query = array()` otherwise it throws this warning:

```
PHP Warning:  Declaration of EDD\Database\Queries\Order::query($query) should be compatible with EDD\Database\Query::query($query = Array) in C:\wamp\www\edd3\wp-content\plugins\easy-digital-downloads\includes\database\queries\class-order.php on line 324
```

Reply:
Weird. For some reason I am not getting those despite WP_DEBUG being true. Do you know if I need a specific version of PHP or a setting to have that show?

Reply:
I _think_ they got more strict about this in PHP 7.0+.